### PR TITLE
DM-43563: Add flat correction placeholder.

### PR DESCRIPTION
**Pull request by aferte**

### CODE REVIEW

File: python/lsst/ip/isr/isrTaskLSST.py [hunk 1]
```diff
@@ -404,6 +404,13 @@ class IsrTaskLSSTConfig(pipeBase.PipelineTaskConfig,
         default=True,
     )
 
+    # Flat correction.
+    doFlat = pexConfig.Field(
```
Comment (plazas) [inline python/lsst/ip/isr/isrTaskLSST.py:408]: @erykoff has mentioned that we are going to have multiple types of flats, not just one single flat. Does this placeholder refer to a particular flat?  is this the dome flat? Should we add comments to the code?

File: python/lsst/ip/isr/isrTaskLSST.py [hunk 2]
```diff
@@ -1320,6 +1329,11 @@ def run(self, *, ccdExposure, dnlLUT=None, bias=None, deferredChargeCalib=None,
             bfKernelOut, bfGains = self.getBrighterFatterKernel(detector, bfKernel)
             ccdExposure = self.applyBrighterFatterCorrection(ccdExposure, flat, dark, bfKernelOut, bfGains)
 
+        if self.config.doFlat:
+            # Input units: electrons
+            self.log.info("Applying flat correction.")
+            self.flatCorrection(ccdExposure, flat)
```
Comment (plazas) [inline python/lsst/ip/isr/isrTaskLSST.py:1335]: This `flatCorrection` method has not been implemented yet? Will there be future tickets to do so, and to add testing? If so, perhaps we could add their numbers here as comments.

**Review by plazas — APPROVED** [review]
I think it looks good as long as this is a "transition" ticket, i.e., there will be follow-up tickets for testing and making sure that things run properly. I think that's the case, so I'm just asking for some clarification in the comments.

File: python/lsst/ip/isr/isrTaskLSST.py (hunk 2 above)
Comment (aferte) [reply python/lsst/ip/isr/isrTaskLSST.py:1335]: Ah good catch - thanks. For the moment we will use the same as for the original task, which I meant to copy over.

File: python/lsst/ip/isr/isrTaskLSST.py (hunk 1 above)
Comment (aferte) [reply python/lsst/ip/isr/isrTaskLSST.py:408]: I am adding a comment to say it would correspond to a background flat in this placeholder.

**Pull request merged**